add rule-color-no-hex

makotot opened the pull request:
ref : #425.
Simply added by imitating `color-no-invalid-hex`.

davidtheclark commented on src/rules/color-no-hex/__tests__/index.js:
```diff
@@ -0,0 +1,69 @@
+
+  tr.ok("a { color: pink; }")
+  tr.ok("a { color: rgba(0, 0, 0, 0); }")
+  tr.ok("a { color: #12345; }")
```
I'd think that if I turned this rule on I would not want to also have to turn on `color-no-invalid-hex` --- so the *apparent attempts* at hex values should be flagged just as much as the valid hex values. What do you think about that?

davidtheclark commented on src/rules/color-no-hex/__tests__/index.js:
```diff
@@ -0,0 +1,69 @@
+    column: 23,
+  })
+
+  // 8 digits
```
Looks like these comments are reversed.

davidtheclark commented on src/rules/color-no-hex/index.js:
```diff
@@ -0,0 +1,44 @@
+        if (!/[:,\s]/.test(declString[match.startIndex - 1])) { return }
+
+        const hexMatch = /^#[0-9A-Za-z]+/.exec(declString.substr(match.startIndex))
+        if (!hexMatch) { return }
```
As mentioned above, I'm of the opinion that this rule should check *apparent attempts* at hex values, not just legit hex values.

davidtheclark commented:
Thanks @makotot!

I made a few comments. Wonder what you think about them (and any other @stylelint/owners).

Also, if you want to polish this PR off you could 
- add a note to the CHANGELOG (under `## Head` at the top)
- add this rule's README to the rule index in the documentation
... and I think that would be it.

makotot commented:
@davidtheclark 
Thanks for quick review :) I'll check later.

jeddy3 commented:
@makotot Thanks for contributing :)

> I'd think that if I turned this rule on I would not want to also have to turn on color-no-invalid-hex --- so the apparent attempts at hex values should be flagged

Sounds good to me.

makotot commented:
Updated.

> As mentioned above, I'm of the opinion that this rule should check apparent attempts at hex values, not just legit hex values.

I agree with your opinion. In real use scene, I would not want to allow any hex values that are incorrect.

> Also, if you want to polish this PR off you could
> 
> add a note to the CHANGELOG (under ## Head at the top)
> add this rule's README to the rule index in the documentation

Thanks for letteing me know :) I also updated CHANGELOG and this rule's README.

davidtheclark commented:
Thanks @makotot!
